src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java

```diff
@@ -341,10 +314,10 @@ public void onEntityInteractSpecific(PlayerInteractEvent.EntityInteractSpecific
 		else {
 			boolean isSmall = armorStand.isSmall();
 			Vec3d vec = event.getLocalPos();
-			double d4 = isSmall ? vec.yCoord * 2.0D : vec.yCoord;
+			double d4 = isSmall ? vec.y * 2.0D : vec.y;
 			ItemStack stack = event.getItemStack();
 
-			if (stack == null || !(stack.getItem() instanceof ItemElytra) || event.getHand() != EnumHand.MAIN_HAND) {
+			if (stack.isEmpty() || !(stack.getItem() instanceof ItemElytra) || event.getHand() != EnumHand.MAIN_HAND) {
```

Comment:
How certain are you that `event.getItemStack()` will never return null?

Reply:
Itemstacks can NEVER be null anymore.

Reply:
Itemstacks cant be null now.

and

```public ItemStack getItemStackFromSlot(EntityEquipmentSlot slotIn)
    {
        if (slotIn == EntityEquipmentSlot.MAINHAND)
        {
            return this.inventory.getCurrentItem();
        }
        else if (slotIn == EntityEquipmentSlot.OFFHAND)
        {
            return this.inventory.offHandInventory.get(0);
        }
        else
        {
            return slotIn.getSlotType() == EntityEquipmentSlot.Type.ARMOR ? (ItemStack)this.inventory.armorInventory.get(slotIn.getIndex()) : ItemStack.EMPTY;
        }
    }```

Reply:
Nice!